test/jest-e2e-tests/publishedData.e2e-spec.ts

```diff
@@ -0,0 +1,139 @@
+        publicationYear: 2020,
+        resourceType: "raw",
+      },
+      datasetPids: [],
+    };
+    await request(app.getHttpServer())
+      .post("/api/v4/PublishedData")
+      .send(publishedData)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${token}` })
+      .expect(201)
+      .expect("Content-Type", /json/)
+      .then((res) => (doi = encodeURIComponent(res.body["doi"])));
+  });
+
+  afterAll(async () => {
+    if (mongoConnection.db) await mongoConnection.db.dropDatabase();
+    await app.close();
+  });
+
+  it("Should register this new published data", async () => {
+    const mockAxiosResponse: AxiosResponse = {
+      data: {},
+      status: 200,
+      statusText: "OK",
+      headers: {},
+      config: {},
+    };
+    jest.spyOn(httpService, "request").mockReturnValue(of(mockAxiosResponse));
+    await request(app.getHttpServer())
+      .post("/api/v4/PublishedData/" + doi + "/register")
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${token}` })
+      .expect(201)
+      .expect("Content-Type", /json/);
+  });
+
+  it("Should fetch this new published data", async () => {
+    await request(app.getHttpServer())
+      .get("/api/v4/PublishedData/" + doi)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${token}` })
+      .expect(200)
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        expect(res.body.status).toEqual("registered");
+      });
+  });
+
+  it("Should not be able to delete published data in registrated", async () => {
+    let archiveManagerToken: string;
+    await request(app.getHttpServer())
+      .post("/api/v3/auth/login")
+      .send({
+        username: "archiveManager",
+        password: "6d3b76392e6f41b087c11f8b77e3f9de",
+      })
+      .set("Accept", "application/json")
+      .expect(201)
+      .then((response) => (archiveManagerToken = response.body.access_token));
+    await request(app.getHttpServer())
+      .delete("/api/v4/PublishedData/" + doi)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${archiveManagerToken}` })
+      .expect(400)
+      .expect("Content-Type", /json/);
+  });
+
+  it("Should amend this new registered published data", async () => {
+    await request(app.getHttpServer())
+      .post("/api/v4/PublishedData/" + doi + "/amend")
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${token}` })
+      .expect(201)
+      .expect("Content-Type", /json/);
+  });
+
+  it("Should fetch this amended published data", async () => {
+    await request(app.getHttpServer())
+      .get("/api/v4/PublishedData/" + doi)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${token}` })
+      .expect(200)
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        expect(res.body.status).toEqual("amended");
+      });
```

Comment:
I'm trying to understand how it returns body or 201 without mocked response

Reply:
but httpServer is mocked only in 1 test, not for all of them